### ISSUE TITLE
Implement TimeSeriesBisectingKMeans

### ISSUE THREAD

**Issue by adagrad** [tslearn-team/tslearn#420]
**Is your feature request related to a problem? Please describe.**
Classical hierarchical clustering approaches requiring a distance matrix are far too resource intense for a lot of samples. Thus scikit learn has introduced the [BisectingKMeans](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.BisectingKMeans.html). However, there I cant use the dtw distance metric

**Describe the solution you'd like**
It would be really great to more or less copy the scikit implementation following the 
TimeSeriesKMeans API. PS If you are really nice you could add an easier way to access the hierarchy as scikit does.

**Comment by nilesh05apr** [comment]
@adagrad i would like to work on BisectingKMeans algorithm implementation same as scikit learn has.

**Comment by rtavenar** [comment]
@nilesh05apr great!

Feel free to open a PR and ask for help at any point!
